components/api/src/main/java/com/hotels/styx/api/extension/service/HealthCheckConfig.java

```diff
@@ -20,7 +20,7 @@
 import java.util.Optional;
 import java.util.concurrent.TimeUnit;
 
-import static com.google.common.base.Objects.toStringHelper;
+import static com.google.common.base.MoreObjects.toStringHelper;
```

Comment:
Ideally we wouldn't use Guava for anything. It causes problems with plugins. Now when you are at it, perhaps replace the Guava based toString with IntelliJ generated, etc.

Furthermore this 300 lines of code looks like an ordinary data class. With Kotlin it would be perhaps 30 lines max.

Reply:
This is PR is intended to just be small incremental changes. I think we should create an issue for larger changes like removing Guava, porting to Kotlin etc.